### PR TITLE
Integrate project tooling

### PR DESCRIPTION
**Pull request by rwaldron**
This toolset is identical to the toolset used on t2-cli, t2-firmware & tessel-av (t2-compiler has the same, but the servr isn't written yet, so it's not in use).

Currently, almost nothing passes.

- grunt jshint: fail; all files need to be cleaned up to pass
- grunt jscs: fail; all files need to be cleaned up to pass
- grunt nodeunit: pass; there is only one test, @wyze will be implementing more

Signed-off-by: Rick Waldron <waldron.rick@gmail.com>

### CODE REVIEW

**Comment by wyze** [comment]
Awesome, thanks! After this gets merged can status checks be enabled on this repo so it runs Travis/Appveyor on PRs?

LGTM (If my vote counts!)

File: test/common/bootstrap.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+global.Attiny = require('./attiny-common-mock');
+global.exported = require('../../index');
+
+
+global.EventEmitter = events.EventEmitter;
+global.sandbox = global.sinon.sandbox.create();
+global.Ambient = exported.Ambient;
+
+// Notes:
+// - `exported` will likely be usefule for stubbing `use`
```
Comment (wyze) [inline test/common/bootstrap.js]: Misspelled useful here.

Comment (rwaldron) [inline test/common/bootstrap.js]: Of course I did hahaha

**Comment by rwaldron** [comment]
@johnnyman727 if you could just give this a once-over, @wyze and I would be grateful :)

**Comment by johnnyman727** [comment]
Looks good to me besides the typo. All the commands work as expected.

@wyze If you run into a ton of incompat issues on T1, we may want to consider jumping the major or minor version and leaving T1 compat in the dust (with a note in the README to install @ that last version).

**Comment by wyze** [comment]
Ping @rwaldron to fix his typo so this can be merged. :)

**Comment by rwaldron** [comment]
Merged https://github.com/tessel/ambient-attx4/commit/59f12d13c76a6937dddb3151397c6cc044f82b5f

**Comment by rwaldron** [comment]
@wyze take it away!

**Pull request closed without merging**